tools/Makefile.unix

```diff
@@ -455,28 +462,28 @@ pass2dep: context tools/mkdeps$(HOSTEXEEXT) tools/cnvwindeps$(HOSTEXEEXT)
 # file in the NuttX tools GIT repository for additional information.
 
 config: apps_preconfig
-	$(Q) APPSDIR=${CONFIG_APPS_DIR} kconfig-conf Kconfig
+	$(Q) APPSDIR=${CONFIG_APPS_DIR} EXTERNALDIR=$(EXTERNALDIR) kconfig-conf Kconfig
 
 oldconfig: apps_preconfig
-	$(Q) APPSDIR=${CONFIG_APPS_DIR} kconfig-conf --oldconfig Kconfig
+	$(Q) APPSDIR=${CONFIG_APPS_DIR} EXTERNALDIR=$(EXTERNALDIR) kconfig-conf --oldconfig Kconfig
 
 olddefconfig: apps_preconfig
-	$(Q) APPSDIR=${CONFIG_APPS_DIR} kconfig-conf --olddefconfig Kconfig
+	$(Q) APPSDIR=${CONFIG_APPS_DIR} EXTERNALDIR=$(EXTERNALDIR) kconfig-conf --olddefconfig Kconfig
```

Comment:
Can we apply PR https://github.com/apache/incubator-nuttx/pull/1785 to APPSDIR and EXTERNALDIR?

Reply:
These are environment variables expected to be received by Kconfig only. Don't confuse with APPDIR which indeed is received by the build system.

Reply:
CONFIG_APPS_DIR/APPDIR is used frequently as TOPDIR, so it make sense to export APPDIR if we export TOPDIR. But I amn't sure for EXTERNALDIR.

Reply:
Maybe I could add "export APPDIR" but inside apps repo. I will look if it makes sense to export it at the top-level of nuttx build (technically it should not be necessary). EXTERNALDIR is the environment variable only passed to Kconfig, these lines are the only place where they are passed around so no need for the export.

Reply:
I just checked: APPDIR is defined once per Makefile.unix/win and then just passed into apps make, so I think it is cleaner to only globally expose APPDIR inside apps repo. I will make a PR for it later.